lib/bolt/pal.rb

```diff
@@ -136,33 +145,35 @@ def in_bolt_compiler
       # TODO: If we always call this inside a bolt_executor we can remove this here
       setup
       r = Puppet::Pal.in_tmp_environment('bolt', modulepath: @modulepath, facts: {}) do |pal|
-        pal.with_script_compiler do |compiler|
-          alias_types(compiler)
-          register_resource_types(Puppet.lookup(:loaders)) if @resource_types
-          begin
-            Puppet.override(yaml_plan_instantiator: Bolt::PAL::YamlPlan::Loader) do
-              yield compiler
-            end
-          rescue Bolt::Error => e
-            e
-          rescue Puppet::DataBinding::LookupError => e
-            if e.issue_code == :HIERA_UNDEFINED_VARIABLE
-              message = "Interpolations are not supported in lookups outside of an apply block: #{e.message}"
-              PALError.new(message)
-            else
-              PALError.from_preformatted_error(e)
-            end
-          rescue Puppet::PreformattedError => e
-            if e.issue_code == :UNKNOWN_VARIABLE &&
-               %w[facts trusted server_facts settings].include?(e.arguments[:name])
-              message = "Evaluation Error: Variable '#{e.arguments[:name]}' is not available in the current scope "\
-                        "unless explicitly defined. (file: #{e.file}, line: #{e.line}, column: #{e.pos})"
-              PALError.new(message)
-            else
+        Puppet.override(bolt_project: @project) do
```

Comment:
We should be able to include `yaml_plan_instantiator` in this same `override` call so we can avoid a level of nesting.